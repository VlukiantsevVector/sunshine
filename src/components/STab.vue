<template>
  <router-link :class="classes('link')" :to="to"><slot /></router-link>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
  name: 's-tab',

  props: {
    to: String,
  },

  watch: {
    $route: {
      handler(route) {
        if (route.path === this.to) {
          (this.$parent as any).setActiveTab(this);
        }
      },
      immediate: true,
    },
  },
});
</script>
