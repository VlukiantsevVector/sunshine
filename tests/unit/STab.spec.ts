import { mount, RouterLinkStub } from '@vue/test-utils';
import STab from '@/components/STab.vue';

describe('STab', () => {
  it('renders slot content', () => {
    const wrapper = mount(STab, {
      propsData: {
        to: '?test',
      },
      slots: {
        default: 'Tab 1',
      },
      stubs: {
        RouterLink: RouterLinkStub,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('has correct routerlink', () => {
    const link = '?test';
    const wrapper = mount(STab, {
      propsData: {
        to: '?test',
      },
      stubs: {
        RouterLink: RouterLinkStub,
      },
    });

    expect(wrapper.find(RouterLinkStub).props().to).toBe(link);
  });
});
