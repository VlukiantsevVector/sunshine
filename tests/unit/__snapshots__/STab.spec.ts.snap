// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`STab renders slot content 1`] = `<a class="s-tab__link"><span tabindex="-1" class="s-tab__wrapper"><span class="s-tab__title">Tab 1</span></span></a>`;
